File: dsa/typescript/insertionSort.ts
let list:number[] = [22, 18, 87, 13, 1, 89, 43, 51, 77, 93];
let newList:number[] = insertionSort(list);
console.log(newList);

function insertionSort(list:number[]):number[]{
    for(let i=1; i<list.length; i++){
        let curr:number = list[i];
        let u;
        for(u=i-1; u >= 0 && list[u] > curr; u--){
            list[u+1] = list[u];
        }

        list[u+1] = curr;
    }
    return list;
}

File: dsa/go/insertionSort.go
package main

import (
	"fmt"
)

func main() {
	list := []int{23, 1, 56, 90, 56, 43, 75}
	newList := insertionSort(list)
	for _, v := range newList {
		fmt.Println(v)
	}
}

func insertionSort(list []int) []int {
	for i := 1; i < len(list); i++ {
		u := i - 1
		curr := list[i]
		for u >= 0 && list[u] > curr {
			list[u+1] = list[u]
			u--
		}
		list[u+1] = curr
	}
	return list
}
